File: apps/mobile/src/api/homesApi.ts
import { apiClient } from './apiClient';

export interface HomeResponse {
  id: string;
  name: string;
  ownerId: string;
  countryCode: string;
  timezone: string;
}

export const homesApi = {
  async createHome(payload: { name: string; countryCode: string; timezone?: string }): Promise<HomeResponse> {
    const { data } = await apiClient.post<HomeResponse>('/homes', payload);
    return data;
  },
  async getMyHomes(): Promise<HomeResponse[]> {
    const { data } = await apiClient.get<HomeResponse[]>('/homes/mine');
    return data;
  },
};

File: apps/hub/app/src/main/kotlin/cl/babyguardian/hub/data/api/HomesApi.kt
package cl.babyguardian.hub.data.api

import cl.babyguardian.hub.data.model.CreateHomeRequest
import cl.babyguardian.hub.data.model.HomeDto
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface HomesApi {
    @GET("homes/mine")
    suspend fun getMyHomes(@Header("Authorization") authorization: String): List<HomeDto>

    @POST("homes")
    suspend fun createHome(
        @Header("Authorization") authorization: String,
        @Body request: CreateHomeRequest,
    ): HomeDto
}
